### DISCUSSION TITLE
working directory for interactive quarto document in R

### DISCUSSION THREAD

**Discussion by AmelZulji** [quarto-dev/quarto-cli#4752] [Q&A]
Dear community, 

I am following this post here https://github.com/rstudio/rstudio/issues/11361 to honor working directory for interactive qmd document. However, seems like `execute-dir` is completely unresponsive. When document is rendered `getwd()` shows the directory in which the files is. However, interactively  `getwd()` shows the project root directory (it does not change with changing `execute-dir` to file or project). 

Is it possible that the `execute-dir` is unresponsive because I am working with qmd document within .rproject and not quarto project? If yes, is there a workaround?

Regards,
Amel

**Comment by EinMaulwurf** [comment]
You can try settting `Settings -> R Markdown -> Evaluate Chunks in Directory: Project` in RStudio. For me, this helps with quarto markdown (qmd) files as well.

**Comment by cderv** [comment]
> working directory for interactive qmd document.

Can you give an example of your current setup so that we know for sure what you are trying ? 

> Is it possible that the execute-dir is unresponsive because I am working with qmd document within .rproject and not quarto project? If yes, is there a workaround?

For `execute-dir` configuration, if you are refering to this one: https://quarto.org/docs/projects/code-execution.html#working-dir, then this is a Quarto project option. So if you are not using quarto project with a `_quarto.yml`, I don't know where you set your configuration but this won't work. 

Are you in this situation: https://github.com/rstudio/rstudio/issues/11361#issuecomment-1212562033 ? 

`execute-dir` is not an option for a document, this is really for project. 

If you are looking for the document level option, then this is at knitr level with Package option (https://yihui.org/knitr/options/#package-options) using `root.dir`. They can be set through YAML too https://quarto.org/docs/computations/r.html#knitr-options

However, beware playing with changing the execution dir. This could lead to undesired side effect in a project or other process. Usually, executing code content inside the path where the document is located is enough, and when needed to access resources elsewhere, playing with path construction is better. 

Anyhow, this should help
